deps/chakrashim/src/jsrtcontextshim.cc

```diff
@@ -447,6 +451,46 @@ bool ContextShim::ExecuteChakraShimJS() {
                         &result) == JsNoError;
 }
 
+bool ContextShim::ExecuteChakraDebugShimJS(JsValueRef * chakraDebugObject) {
+  wchar_t buffer[_countof(chakra_debug_native) + 1];
+
+  if (StringConvert::CopyRaw<unsigned char, wchar_t>(chakra_debug_native,
+    _countof(chakra_debug_native),
+    buffer,
+    _countof(chakra_debug_native)) != JsNoError) {
+    return false;
+  }
+
+  // Ensure the buffer is null terminated
+  buffer[_countof(chakra_debug_native)] = L'\0';
+
+  JsValueRef getInitFunction;
+  if (JsParseScriptWithAttributes(buffer,
+    JS_SOURCE_CONTEXT_NONE,
+    L"chakra_debug.js",
+    JsParseScriptAttributeLibraryCode,
+    &getInitFunction) != JsNoError) {
+    return false;
+  }
+
+  JsValueRef initFunction;
+  if (CallFunction(getInitFunction, &initFunction) != JsNoError) {
+    return false;
+  }
+
+  JsValueRef traceDebugJsonRef;
+  JsBoolToBoolean(v8::g_trace_debug_json, &traceDebugJsonRef);
+
+  JsValueRef arguments[] = { this->globalObject, this->globalObject,
+      this->keepAliveObject, traceDebugJsonRef };
+  JsErrorCode errorCode = JsCallFunction(initFunction, arguments,
+      _countof(arguments), chakraDebugObject);
+
+  CHAKRA_VERIFY_NOERROR(errorCode);
+
+  return errorCode == JsNoError;
```

Comment:
>rrorCode == JsNoError; [](start = 10, length = 22)

nit: won't this always be true given that you have CHAKRA_VERIFY_NOERROR(errorCode); above?